MTA-6893 Document Hub OIDC authentication (#382)
migtools/mta-documentation: docs/topics/mta-install/con_administrative-tasks.adoc

:_newdoc-version: 2.18.5
:_template-generated: 2025-07-31
:_mod-docs-content-type: CONCEPT

[id="administrative-tasks_{context}"]
= Administrative tasks

[role="_abstract"]

> Administrators can configure user authentication and understand user, role, and token management. {ProductFullName} offers you the flexibility to customize configuration as you need.

Reviewer: ```suggestion
As an administrator, you can configure user authentication and manage users, roles, and tokens to customize your environment.
```

> You can use any one of the authentication flows by using {ProductShortName} Hub OpenID Connect (OIDC):

Reviewer: ```suggestion
You can use any of the following authentication flows with the {ProductShortName} Hub OpenID Connect (OIDC) provider:
```

Local authentication and authorization::

> Add users and link roles to them in the {ProductShortName} application. You do not need any extra configuration.

Reviewer: ```suggestion
Local authentication and authorization::
You can add users and link roles to them in the {ProductShortName} application without any extra configuration.
```

Lightweight Directory Access Protocol (LDAP)-backed authentication::

> Configure an `LdapProvider` CR to integrate LDAP users. Optionally, configure and deploy secrets in your Red Hat OpenShift cluster for a secure connection with the LDAP client.

Reviewer: ```suggestion
Configure an `LdapProvider` custom resource (CR) to integrate LDAP users. Optionally, configure and deploy secrets in your Red Hat OpenShift Container Platform cluster for a secure connection with the LDAP client.
```

Author: Hey Andy,
Thanks for the review! In an old PR, we were asked to write Red Hat OpenShift since OCP is one of the options customers opt for. Please let me know your thoughts on this.

Reviewer: Not a problem... as i said there is nothing majorly wrong with anything you have done.
if you did not make the changes i would merge and ask you to do a polish post-GA

Federated authentication to Identity Providers (IdPs)::

> Configure the `IdentityProvider` CR to federate authentication to an IdP such as Google, Okta, Keycloak, or Microsoft Entra ID (formerly, Azure Active Directory). Optionally, configure and deploy secrets in your Red Hat OpenShift cluster for a secure connection with the IdP client.

Reviewer: ```suggestion
Configure the `IdentityProvider` CR to federate authentication to an IdP such as Google, Okta, Keycloak, or Microsoft Entra ID (formerly, Azure Active Directory). Optionally, configure and deploy secrets in your Red Hat OpenShift Container Platform cluster for a secure connection with the IdP client.
```

> To migrate Keycloak implemented in previous versions after an {ProductShortName} 8.2.0 upgrade, see Keycloak migration.

Reviewer: ```suggestion
To migrate Keycloak instances that were implemented in earlier versions after you upgrade to {ProductShortName} 8.2.0, see Keycloak migration.
```

[id="optional-configuration_{context}"]
== Optional configurations

> If necessary, you can configure authentication fields, custom roles, and API keys.

Reviewer: ```suggestion
You can perform the following optional configuration tasks:
```

* Configure authentication and token variables in the Tackle custom resource (CR).
* Create custom roles.

> * Create and revoke API keys.

Reviewer: ```suggestion
* Configure authentication and token variables in the Tackle custom resource (CR).
* Create custom roles.
* Create and revoke application programming interface (API) keys.
```
